packages/filebrowser/src/listing.ts

```diff
@@ -514,6 +514,21 @@ export class DirListing extends Widget {
       });
   }
 
+  /**
+   * Select all directory listing items
+   */
+  selectAll(): void {
+    const items = this._model.items();
+    const newSelection: { [key: string]: boolean } = {};
+
+    for (const item of items) {
+      newSelection[item.path] = true;
+    }
+
+    this.selection = newSelection;
```

Comment:
Following https://github.com/jupyterlab/jupyterlab/pull/17563 this should also emit `selectionChanged` signal:

```suggestion
    this.selection = newSelection;
    this._selectionChanged.emit()
```